watch/watch.py

```
@@ -132,9 +142,12 @@ def stream(self, func, *args, **kwargs):
                     if self._stop:
                         break
             finally:
-                kwargs['resource_version'] = self.resource_version
                 resp.close()
                 resp.release_conn()
+                if self.resource_version is not None:
+                    kwargs['resource_version'] = self.resource_version
+                else:
```

Do you need the else: break here?

---

Yes. The log streaming does not have `resource_version`. So once the response finishes, you have to stop.

---

In fact that was not good because using break inside finally swallows up the exception. I change it to setting a flag.